### PR TITLE
Jkantor/waiting bug

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** Fix a bug where the waiting details page would not include legitimate 'waiting' students. 

JIRA: [AU-21](https://openedx.atlassian.net/browse/AU-21)

**What changed?**

- changed `get_waiting_step_details ` to not include "unfinished" (aka, no actual assessment) PeerWorkflowItems
- I also did a small refactor to move some "magic strings" from the assessment/api files to their own file, to avoid a circular import
- some quality errors that somehow slipped through?

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

I won't lie, this is gonna be a pain to test.
- Make an ORA with only a peer review step, with one required grade and one required graded_by, in a course with at least two learners
- Make a submission with both learners.
- For learner 1, submit an assessment for learner 2
- Learner 1 should be in waiting because they have not received a peer review
- Make sure learner 2 has been "assigned" learner 1, by which I mean log in as learner 2 and make sure that learner 1's submission is listed as your active review in your peer review step
- Go to the instructor dashboard. There should be one learner listed as "peer" (learner 2) and one learner listed as "waiting"  (learner 1).
- Click the "1" in waiting to be brought to the dashboard.
-  NOTE: IF THE NUMBER UNDER WAITING IS NOT CLICKABLE, YOU MUST MAKE THE FOLLOWING CHANGE:
   In edx-platform/lms/envs/devstack.py, add the following line at the end of the file: 
   `FEATURES['ENABLE_XBLOCK_VIEW_ENDPOINT'] = True` 
- No learners should be listed in the tool.
- Switch to this branch and reload the waiting learner tool. Learner 1 should now be listed.
 
**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/assessment/test/test_peer.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ def test_get_waiting_step_details(self):
         self._create_student_and_submission("Tim", "Tim's answer")
         bob_sub, bob = self._create_student_and_submission("Bob", "Bob's answer")
         peer_api.get_submission_to_assess(bob_sub['uuid'], 1)
-        assessment = peer_api.create_assessment(
```
Comment (jansenk) [inline openassessment/assessment/test/test_peer.py:221]: this and `test_flexible_peer_grading__zero_graded` changes were quality issues that have somehow slipped in

File: openassessment/assessment/test/test_peer.py [hunk 2]
```diff
@@ -1755,15 +1754,86 @@ def test_flexible_peer_grading__zero_graded(self):
...
+
+        # Call get_submission_to_assess once more so that target_learner has an open incomplete peer assessment
+        peer_api.get_submission_to_assess(target_learner_sub['uuid'], target_learner['student_id'])
+
+        # Call get_submission_to_assess so all five learners in other_learner_submissions are
+        # currently assessing target_learner
+        for sub, student in other_learner_submissions:
+            chosen_submission = peer_api.get_submission_to_assess(sub['uuid'], student['student_id'])
+            self.assertIsNotNone(chosen_submission)
+            self.assertEqual(chosen_submission['uuid'], target_learner_sub['uuid'])
```
Comment (jansenk) [inline openassessment/assessment/test/test_peer.py:1805]: This currently will always be true, but the rest of the test depends on it, so i'm asserting here just in case something changes in the future

File: openassessment/assessment/models/peer.py [hunk 3]
```diff
@@ -248,13 +249,31 @@ def get_waiting_step_details(
                 'graded_by': 2
             }
         """
+        Count, Q = models.Count, models.Q
```
Comment (muselesscreator) [inline openassessment/assessment/models/peer.py:252]: What is this doing?

Comment (jansenk) [reply openassessment/assessment/models/peer.py:252]: I didn't want to have to type models.Count and models.Q every time I used them so I just assigned them to these variables

Comment (jansenk) [reply openassessment/assessment/models/peer.py:252]: Looking at my final result it's not super necessary. At some point I had a big chain of these filter objects and it was super ugly. I'll remove.

**Review by leangseu-edx — APPROVED** [review]
LGTM 💯

File: openassessment/assessment/score_type_constants.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+""" Constant strings used to identify what type of score an Assessment is. Used in the 'score_type' field """
+
+PEER_TYPE = "PE"
```
Comment (nsprenkle) [inline openassessment/assessment/score_type_constants.py:3]: :chef_kiss:

**Review by nsprenkle — APPROVED** [review]
Visual review looks beautiful. Haven't verified new functionality however.
